lib/intercept.js

```diff
@@ -50,13 +50,18 @@ let allowNetConnect
  * @example
  * // Enables real requests for url that matches google and amazon
  * nock.enableNetConnect(/(google|amazon)/);
+ * @example
+ * // Enables real requests for url that includes google
+ * nock.enableNetConnect(host => host.includes('google'));
  */
 function enableNetConnect(matcher) {
   // TODO-12.x: Replace with `typeof matcher === 'string'`.
   if (_.isString(matcher)) {
     allowNetConnect = new RegExp(matcher)
   } else if (matcher instanceof RegExp) {
     allowNetConnect = matcher
+  } else if (matcher instanceof Function) {
+    allowNetConnect = { test: matcher }
```

Comment:
I could also make the internal variable a function, but this seemed quite elegant - let me know if you want me to change this.